test/firrtl/lower-types.mlir

```diff
@@ -0,0 +1,65 @@
+// RUN: circt-opt -pass-pipeline='firrtl.circuit(lower-firrtl-types)' %s | FileCheck %s
+
+firrtl.circuit "TopLevel" {
```

Comment:
I think the test cases here are a decent start, but I think I need to spend a little time dressing them up according to the [best practices](https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices).